src/modules/java.rs

```diff
@@ -74,19 +71,24 @@ fn get_java_version(context: &Context, config: &JavaConfig) -> Option<String> {
         output.stdout
     };
 
-    parse_java_version(config.version_format, &java_version)
+    format_java_version(&java_version, config.version_format)
 }
 
-fn parse_java_version(version_format: &str, java_version: &str) -> Option<String> {
+fn format_java_version(java_version: &str, version_format: &str) -> Option<String> {
     let re = Regex::new(JAVA_VERSION_PATTERN).ok()?;
     let captures = re.captures(java_version)?;
     let version = &captures["version"];
 
-    Some(
-        VersionFormatter::new(version_format)
-            .ok()?
-            .format_version(version),
-    )
+    let formatted = VersionFormatter::new(version_format)
+        .and_then(|formatter| formatter.format_version(version));
+
+    match formatted {
+        Ok(formatted) => Some(formatted),
+        Err(error) => {
```

Comment:
What conditions would lead to this warning? I would rather just silently format the raw version than log a warning when this happens.

Reply:
> I would rather just silently format the raw version than log a warning when this happens.

This is _silent_ unless you explicitly opt-in by setting `STARSHIP_LOG` environment variable to the `warn` level.

> What conditions would lead to this warning?

The simplest case that would cause an error is a malformed version format like:
```toml
version = "v${raw" # missing right curly
```

This results in (with logging enabled):
```
[WARN] - (starship::modules::java): Error formating `java` version:
 --> 1:3
  |
1 | v${raw
  |   ^---
  |
  = expected variable_name
```